spec/models/dialog_field_serializer_spec.rb

```diff
@@ -53,20 +54,39 @@
       end
 
       it "serializes the dialog_field with the correct values" do
-        expect(dialog_field_serializer.serialize(dialog_field)).to eq(expected_serialized_values.merge(
-          "resource_action" => "serialized resource action",
-          "values"          => "dynamic values"
-        ))
+        expect(dialog_field_serializer.serialize(dialog_field))
+          .to eq(expected_serialized_values.merge(
+                   "resource_action" => "serialized resource action",
+                   "values"          => "dynamic values"
+          ))
+      end
+
+      it 'returns all attributes of a dialog field' do
+        expect(dialog_field_serializer.serialize(dialog_field, true))
+          .to include(expected_serialized_values.merge(
+                        'id'              => dialog_field.id,
+                        'resource_action' => 'serialized resource action',
+                        'values'          => 'dynamic values'
+          ))
       end
     end
 
     context "when the dialog_field is not dynamic" do
       let(:dynamic) { false }
 
       it "serializes the dialog_field" do
-        expect(dialog_field_serializer.serialize(dialog_field)).to eq(expected_serialized_values.merge(
-          "resource_action" => "serialized resource action"
-        ))
+        expect(dialog_field_serializer.serialize(dialog_field))
+          .to eq(expected_serialized_values.merge(
+                   "resource_action" => "serialized resource action"
+          ))
+      end
+
+      it 'returns all attributes of a dialog field' do
+        expect(dialog_field_serializer.serialize(dialog_field, true))
+          .to include(expected_serialized_values.merge(
+                        'id'              => dialog_field.id,
+                        'resource_action' => 'serialized resource action',
+          ))
```

Comment:
Same thing here, there should be two contexts for the different ways we're calling `#serialize`.